spells.yml

```diff
@@ -1000,6 +1031,26 @@ burning_hands:
     - magus
     - sorcerer
     - wizard
+cadaverous_puppet:
+  casting_time: 1 standard action
+  components: V, S
+  concentration: true
+  description: |-
+    You wrest control of a Medium or smaller corpse or skeleton within range. You control the body for the duration of the spell.
+
+    When you cast this spell, and as a minor action on subsequent turns, you can cause the body to stand up if it is prone and walk up to 20 feet, where it remains standing unless you will otherwise. You can also have the body perform simple tasks as part of this action, such as opening a door or pushing a rock. You can't cause the body to attack in this way, however.
+
+    The body counts as an object. It has 15 hit points, 10 AC, and immunity to necrotic, poison, and psychic damage. It uses the Strength, Dexterity, and Constitution scores it had in life. If the body is reduced to 0 hit points, it is destroyed and the spell ends.
+  duration: Up to 8 hours
+  higher_level: When you cast this spell using a spell slot of 3rd level, you can animate a Large or smaller body. When you cast this spell using a spell slot of 4th level, you can animate a Huge or smaller body. When you cast this spell using a spell slot of 5th level or higher, you can animate a body of any size. The body also has an additional 15 hp for each slot level above 2nd.
+  level: 2
+  name: Cadaverous Puppet
+  range: 60 feet
+  sources:
+    - 2D
+  tags:
+    - arcane
+    - necromancy
```

Comment:
So this spell needs a a buff. My current thinking is:
- Add a minor action cantrip that makes a body stand up, walk 20 feet, and stay standing until the end of the turn. Cantrip means it can be paired with e.g. _bloat burst_ on the same turn, but an 8 hour duration would be a bit much I think.
- Keep this spell with its 8 hour duration + concentration, but make the casting and reuse a free action. This way, it doesn't use your very limited AP, so it's more effective long term in combat. Also, it should let you swap targets, so if your body dies you can replace it without burning another spell slot. 8 hours means you can still use it for fun stuff out of combat like scouting with _eyes of the dead_.

Reply:
I think I'd be fine with those changes.